src/bin/config.rs

```
@@ -0,0 +1,76 @@
+use failure::Error;
+use std::env;
+use std::fs::File;
+use std::io::Read;
+use std::path::Path;
+use toml;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Config {
+    pub fitbit: Option<FitbitConfig>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct FitbitConfig {
+    pub client_id: Option<String>,
+    pub client_secret: Option<String>,
+}
+
+impl FitbitConfig {
+    fn from_env_vars(client_id: &str, client_secret: &str) -> FitbitConfig {
+        let mut fc = FitbitConfig {
+            client_id: None,
+            client_secret: None,
+        };
+
+        if let Ok(id) = env::var(client_id) {
+            fc.client_id = Some(id);
+        };
+        if let Ok(secret) = env::var(client_secret) {
+            fc.client_secret = Some(secret);
+        };
+        fc
+    }
+}
+
+impl Config {
+    /// Load a config from the environment. A config object will be constructed
+    /// from a combination of environment variables and/or config files on disk.
+    /// Environment variables supercede values in files.
+    pub fn load(path: Option<&str>) -> Result<Config, Error> {
+        let mut conf = Config { fitbit: None };
+
+        // try config file
+        match path {
+            // custom path to config file, passed as a flag
+            Some(path) => {
+                //if let Some(found) = try_file(&PathBuf::from(Path::new(pathbuf))) {
+                if let Ok(found) = Config::from_toml_file(path) {
```

The code for reading the config in from a file is the same so I'm wondering if it makes sense to restructure this a bit

```rust
let config_path = match path {
    Some(path) => path,
    None => { ... }
}

let config = Config::from_toml_file(config_path)?;
```

It also seems like this is silently ignoring the case where there is a config file but it is not valid TOML.